src/ui/display.rs

```
@@ -144,7 +153,7 @@ impl DrawTarget for DisplayExclusive {
     where
         I: IntoIterator<Item = Pixel<Self::Color>>,
     {
-        let bpp = self.0.var_screen_info.bits_per_pixel / 8;
+        assert!(self.0.var_screen_info.bits_per_pixel == 32);
```

Instead of a potential panic please return an appropriate error in this case.